crates/topos/src/components/setup/subnet/installer.rs

```diff
@@ -0,0 +1,192 @@
+    #[error("There are no available release")]
+    NoValidRelease,
+    #[error("Invalid release metadata")]
+    InvalidReleaseMetadata,
+    #[error("File io error: {0}")]
+    File(std::io::Error),
+}
+
+/// Calculate expected polygon edge binary name for this platform
+/// By convention it is in the format `polygon-edge-<cpu architecture>-<operating system>`
+fn determine_binary_release_name() -> String {
+    "polygon-edge-".to_string() + std::env::consts::ARCH + "-" + std::env::consts::OS
+}
+
+/// Download Polygon Edge binary from repository to requested target directory
+async fn download_binary(file_name: &str, uri: &str, target_directory: &str) -> Result<(), Error> {
+    info!(
+        "Downloading binary `{}` to target directory: {}",
+        file_name, target_directory
+    );
+
+    let response = reqwest::get(uri).await.map_err(Error::Http)?;
+
+    let download_file_name = (target_directory.to_string() + "/" + file_name).replace("\\\\", "\\");
+    let download_file_path = Path::new(&download_file_name);
+    let mut target_file = match File::create(download_file_path) {
+        Err(e) => {
+            error!("Unable to create file: {e}");
+            return Err(Error::File(e));
+        }
+        Ok(file) => file,
+    };
+
+    target_file
+        .write_all(response.bytes().await.map_err(Error::Http)?.as_ref())
+        .map_err(Error::File)?;
+
+    // Set execution rights
+    if let Err(e) = std::process::Command::new("chmod")
+        .arg("u+x")
+        .arg(download_file_name)
+        .output()
+    {
+        error!("Unable to set execution rights for downloaded file: {e}");
+    }
+
+    Ok(())
+}
+
+struct PolygonEdgeRelease {
+    version: String,
+    binary: String,
+    download_url: String,
+}
+
+async fn get_available_releases(repository: &str) -> Result<Vec<PolygonEdgeRelease>, Error> {
+    // Retrieve list of releases
+    let uri = GITHUB_REPO_API.to_string() + repository + "/releases";
+
+    info!("Retrieving Polygon Edge release list {uri}");
+    let client = reqwest::Client::new();
+    let body = client
+        .get(&uri)
+        .header(reqwest::header::USER_AGENT, "Topos CLI")
+        .send()
+        .await
+        .map_err(Error::Http)?
+        .text()
+        .await
+        .map_err(Error::Http)?;
+
+    let body: Vec<Value> = match serde_json::from_str(&body) {
+        Ok(v) => v,
+        Err(e) => {
+            error!("Error parsing release list response: {e}");
+            return Err(Error::InvalidJson(e));
+        }
+    };
+
+    if body.is_empty() {
+        error!("There is no valid Polygon Edge release available");
+        return Err(Error::NoValidRelease);
+    }
+
+    let mut releases: Vec<PolygonEdgeRelease> = Vec::new();
+    // Parse all releases
+    // List of retrieved releases is already sorted, latest release being
+    // the first one in the list
+    for release in &body {
+        let tag_name = release
+            .get("name")
```

Comment:
Checking on https://api.github.com/repos/topos-network/polygon-edge/releases
I believe you want the field `tag_name` instead no?

Reply:
No this is actual name of the release displayed as Headline in release page. `tag_name` is git tag